### PR TITLE
feat(defi): idle multi-account

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR brings multi-account support for Idle:

- removes `chainAdapter.getBIP44Params()` calls
- uses `selectBIP44ParamsByAccountId` selector to derive store BIP44Params from the currently selected `AccountId`
- following https://github.com/shapeshift/web/pull/3241, uses maybe undefined `AccountId` vs. a previously nullable one so we don't have to resort to an unsafe nullish coalescing to an empty string in `accountFilter`

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/lib/issues/943

## Risk

Idle hasn't been tested in quite some time, so this assumes Idle is working with the two stacked PRs underneath this one

## Testing

- Idle Overview / deposit / withdraw modals correctly show the right aggregated / account balances
- Switching accounts is propagated all the way through the modal, including displayed amounts and broadcasted Tx from the currently selected account

### Engineering

☝🏽 

### Operations

☝🏽 

#### Account 0

image
image
image

#### Account 1

image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3248** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3248" target="_blank">Graphite</a> 
    * **PR #3253** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3253" target="_blank">Graphite</a> 
      * **PR #3261** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3261" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3261?utm_source=stack-comment).

File: src/features/defi/providers/idle/components/IdleManager/Deposit/components/Confirm.tsx [hunk 1]
```diff
@@ -87,12 +90,12 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
       const idleOpportunity = await idleInvestor.findByOpportunityId(
         state.opportunity?.positionAsset.assetId ?? '',
       )
-      const bip44Params = chainAdapter.getBIP44Params({ accountNumber: 0 })
       if (!idleOpportunity) throw new Error('No opportunity')
       const tx = await idleOpportunity.prepareDeposit({
         address: state.userAddress,
         amount: bnOrZero(state.deposit.cryptoAmount).times(`1e+${asset.precision}`).integerValue(),
       })
+      console.log({ bip44Params })
```
Comment (gomesalexandre) [inline src/features/defi/providers/idle/components/IdleManager/Deposit/components/Confirm.tsx:98]:
```suggestion
```

File: src/features/defi/providers/idle/components/IdleManager/Deposit/components/Deposit.tsx [hunk 2]
```diff
@@ -54,22 +54,25 @@ export const Deposit: React.FC<DepositProps> = ({
   const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
 
   // user info
-  const balance = useAppSelector(state => selectPortfolioCryptoBalanceByAssetId(state, { assetId }))
+  const balance = useAppSelector(state =>
+    selectPortfolioCryptoBalanceByFilter(state, { assetId, accountId }),
```
Comment (0xdef1cafe) [inline src/features/defi/providers/idle/components/IdleManager/Deposit/components/Deposit.tsx:58]: may want to memo this boy too

File: src/features/defi/providers/idle/components/IdleManager/Overview/IdleOverview.tsx [hunk 3]
```diff
@@ -84,7 +87,7 @@ export const IdleOverview: React.FC<IdleOverviewProps> = ({
   const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
   // user info
   const balance = useAppSelector(state =>
-    selectPortfolioCryptoBalanceByAssetId(state, { assetId: vaultTokenId }),
+    selectPortfolioCryptoBalanceByFilter(state, { accountId, assetId: vaultTokenId }),
```
Comment (0xdef1cafe) [inline src/features/defi/providers/idle/components/IdleManager/Overview/IdleOverview.tsx:90]: meeemo

File: src/features/defi/providers/idle/components/IdleManager/Withdraw/components/Withdraw.tsx [hunk 4]
```diff
@@ -54,7 +57,9 @@ export const Withdraw: React.FC<StepComponentProps> = ({ onNext }) => {
   const marketData = useAppSelector(state => selectMarketDataById(state, underlyingAssetId))
 
   // user info
-  const balance = useAppSelector(state => selectPortfolioCryptoBalanceByAssetId(state, { assetId }))
+  const balance = useAppSelector(state =>
+    selectPortfolioCryptoBalanceByFilter(state, { accountId, assetId }),
```
Comment (0xdef1cafe) [inline src/features/defi/providers/idle/components/IdleManager/Withdraw/components/Withdraw.tsx:61]: memo

File: src/features/defi/providers/idle/components/IdleManager/Deposit/components/Deposit.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/features/defi/providers/idle/components/IdleManager/Deposit/components/Deposit.tsx]: d713fea24e984f81494c805eef38b97019021f69

File: src/features/defi/providers/idle/components/IdleManager/Overview/IdleOverview.tsx (hunk 3 above)
Comment (gomesalexandre) [reply src/features/defi/providers/idle/components/IdleManager/Overview/IdleOverview.tsx]: d713fea24e984f81494c805eef38b97019021f69

File: src/features/defi/providers/idle/components/IdleManager/Overview/IdleOverview.tsx [hunk 5]
```diff
@@ -67,9 +70,7 @@ export const IdleOverview: React.FC<IdleOverviewProps> = ({
   const [opportunity, setOpportunity] = useState<IdleOpportunity | null>(null)
```
Comment (gomesalexandre) [inline src/features/defi/providers/idle/components/IdleManager/Overview/IdleOverview.tsx:70]: Added via Giphy

Comment (gomesalexandre) [reply src/features/defi/providers/idle/components/IdleManager/Overview/IdleOverview.tsx]: https://github.com/shapeshift/web/pull/3261/commits/156007fe620a3b44f3f13eb83c7940fd094a5379

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**